From c387e045607b7353bff4d887548b11232d6074bc Mon Sep 17 00:00:00 2001
From: Alberto Ponces <ponces26@gmail.com>
Date: Wed, 9 Feb 2022 12:34:47 +0000
Subject: [PATCH 4/8] treble: Set OTA JSON URL

Change-Id: I8f817b90d42629c208ceb45598daf5293850b953
---
 lineage.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lineage.mk b/lineage.mk
index 172bb01..68b57f4 100644
--- a/lineage.mk
+++ b/lineage.mk
@@ -1,3 +1,6 @@
 $(call inherit-product, vendor/lineage/config/common_full_phone.mk)
 -include vendor/lineage/build/core/config.mk
 -include vendor/lineage/build/core/apicheck.mk
+
+PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
+    ro.system.ota.json_url=https://downloads.sourceforge.net/project/andyyan-gsi/lineage-19.x/ota.json
-- 
2.25.1

