From a7c48516ef7fddb7c71566299daf2bdab72e3bcd Mon Sep 17 00:00:00 2001
From: Andy CrossGate Yan <GeForce8800Ultra@gmail.com>
Date: Tue, 26 Oct 2021 08:16:00 +0000
Subject: [PATCH] [TEMP] Build fixes for Android 12

---
 Android.mk | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/Android.mk b/Android.mk
index d5b099a2f..efc60d309 100644
--- a/Android.mk
+++ b/Android.mk
@@ -51,12 +51,14 @@ LOCAL_SHARED_LIBRARIES += \
     libjni_trueportrait \
     libjni_filtergenerator
 
-LOCAL_REQUIRED_MODULES := libts_detected_face_jni libts_face_beautify_jni
-
 LOCAL_PROGUARD_FLAG_FILES := proguard.flags
 
 LOCAL_JAVA_LIBRARIES += org.apache.http.legacy
 
+LOCAL_OPTIONAL_USES_LIBRARIES := \
+    com.google.android.media.effects \
+    org.apache.http.legacy
+
 LOCAL_JARJAR_RULES := $(LOCAL_PATH)/jarjar-rules.txt
 
 include $(BUILD_PACKAGE)
-- 
2.25.1

