From 1c6ae8ea3a7075d1030d042057ab335817271683 Mon Sep 17 00:00:00 2001
From: Andy CrossGate Yan <GeForce8800Ultra@gmail.com>
Date: Fri, 13 Sep 2024 08:10:14 +0800
Subject: [PATCH 01/45] Add DisplayCutoutEmulationNoneOverlay

Unlike AvoidAppsInCutoutOverlay or NoCutoutOverlay, this overlay simply
removes cutout definitions, making full display area usable
Intended for devices with center cutouts, esp. badly defined ones

Change-Id: I5e95bbd2dabcecbfdf8e25dea68acc86acf8fe48
---
 packages/overlays/Android.bp                  |  1 +
 .../Android.bp                                | 14 +++++++
 .../AndroidManifest.xml                       | 26 ++++++++++++
 .../res/values-zh-rCN/strings.xml             | 21 ++++++++++
 .../res/values/config.xml                     | 40 +++++++++++++++++++
 .../res/values/strings.xml                    | 21 ++++++++++
 6 files changed, 123 insertions(+)
 create mode 100644 packages/overlays/DisplayCutoutEmulationNoneOverlay/Android.bp
 create mode 100644 packages/overlays/DisplayCutoutEmulationNoneOverlay/AndroidManifest.xml
 create mode 100644 packages/overlays/DisplayCutoutEmulationNoneOverlay/res/values-zh-rCN/strings.xml
 create mode 100644 packages/overlays/DisplayCutoutEmulationNoneOverlay/res/values/config.xml
 create mode 100644 packages/overlays/DisplayCutoutEmulationNoneOverlay/res/values/strings.xml

diff --git a/packages/overlays/Android.bp b/packages/overlays/Android.bp
index 5e001fba6aa1..11269cbacb06 100644
--- a/packages/overlays/Android.bp
+++ b/packages/overlays/Android.bp
@@ -25,6 +25,7 @@ phony {
         "DisplayCutoutEmulationCornerOverlay",
         "DisplayCutoutEmulationDoubleOverlay",
         "DisplayCutoutEmulationHoleOverlay",
+        "DisplayCutoutEmulationNoneOverlay",
         "DisplayCutoutEmulationTallOverlay",
         "DisplayCutoutEmulationWaterfallOverlay",
         "FontNotoSerifSourceOverlay",
diff --git a/packages/overlays/DisplayCutoutEmulationNoneOverlay/Android.bp b/packages/overlays/DisplayCutoutEmulationNoneOverlay/Android.bp
new file mode 100644
index 000000000000..a61430bad341
--- /dev/null
+++ b/packages/overlays/DisplayCutoutEmulationNoneOverlay/Android.bp
@@ -0,0 +1,14 @@
+package {
+    // See: http://go/android-license-faq
+    // A large-scale-change added 'default_applicable_licenses' to import
+    // all of the 'license_kinds' from "frameworks_base_license"
+    // to get the below license kinds:
+    //   SPDX-license-identifier-Apache-2.0
+    default_applicable_licenses: ["frameworks_base_license"],
+}
+
+runtime_resource_overlay {
+    name: "DisplayCutoutEmulationNoneOverlay",
+    theme: "DisplayCutoutEmulationNone",
+    product_specific: true,
+}
diff --git a/packages/overlays/DisplayCutoutEmulationNoneOverlay/AndroidManifest.xml b/packages/overlays/DisplayCutoutEmulationNoneOverlay/AndroidManifest.xml
new file mode 100644
index 000000000000..31942fce96e4
--- /dev/null
+++ b/packages/overlays/DisplayCutoutEmulationNoneOverlay/AndroidManifest.xml
@@ -0,0 +1,26 @@
+<!--
+  ~ Copyright (C) 2020 The Android Open Source Project
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~      http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+        package="com.android.internal.display.cutout.emulation.none"
+        android:versionCode="1"
+        android:versionName="1.0">
+    <overlay android:targetPackage="android"
+            android:category="com.android.internal.display_cutout_emulation"
+            android:priority="1"/>
+
+    <application android:label="@string/display_cutout_emulation_overlay" android:hasCode="false"/>
+</manifest>
diff --git a/packages/overlays/DisplayCutoutEmulationNoneOverlay/res/values-zh-rCN/strings.xml b/packages/overlays/DisplayCutoutEmulationNoneOverlay/res/values-zh-rCN/strings.xml
new file mode 100644
index 000000000000..c2440981dfad
--- /dev/null
+++ b/packages/overlays/DisplayCutoutEmulationNoneOverlay/res/values-zh-rCN/strings.xml
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (C) 2020 The Android Open Source Project
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~      http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+   -->
+
+<resources xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <string name="display_cutout_emulation_overlay">"æ— "</string>
+</resources>
diff --git a/packages/overlays/DisplayCutoutEmulationNoneOverlay/res/values/config.xml b/packages/overlays/DisplayCutoutEmulationNoneOverlay/res/values/config.xml
new file mode 100644
index 000000000000..613eaa7dab6a
--- /dev/null
+++ b/packages/overlays/DisplayCutoutEmulationNoneOverlay/res/values/config.xml
@@ -0,0 +1,40 @@
+<!--
+  ~ Copyright (C) 2020 The Android Open Source Project
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~      http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+
+    <!-- The bounding path of the cutout region of the main built-in display.
+         Must either be empty if there is no cutout region, or a string that is parsable by
+         {@link android.util.PathParser}.
+
+         The path is assumed to be specified in display coordinates with pixel units and in
+         the display's native orientation, with the origin of the coordinate system at the
+         center top of the display.
+
+         To facilitate writing device-independent emulation overlays, the marker `@dp` can be
+         appended after the path string to interpret coordinates in dp instead of px units.
+         Note that a physical cutout should be configured in pixels for the best results.
+         -->
+    <!-- Display cutout configuration -->
+    <string translatable="false" name="config_mainBuiltInDisplayCutout" />
+
+    <string translatable="false" name="config_mainBuiltInDisplayCutoutRectApproximation" />
+
+    <!-- Whether the display cutout region of the main built-in display should be forced to
+         black in software (to avoid aliasing or emulate a cutout that is not physically existent).
+     -->
+    <bool name="config_fillMainBuiltInDisplayCutout">false</bool>
+</resources>
diff --git a/packages/overlays/DisplayCutoutEmulationNoneOverlay/res/values/strings.xml b/packages/overlays/DisplayCutoutEmulationNoneOverlay/res/values/strings.xml
new file mode 100644
index 000000000000..382c67cfec13
--- /dev/null
+++ b/packages/overlays/DisplayCutoutEmulationNoneOverlay/res/values/strings.xml
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (C) 2020 The Android Open Source Project
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~      http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+
+    <string name="display_cutout_emulation_overlay">None</string>
+
+</resources>
-- 
2.34.1

